homeassistant/components/camera/synology.py

```diff
@@ -203,48 +93,348 @@ def camera_image(self):
-                response = yield from self._websession.get(
-                    image_url,
-                    params=image_payload
-                )
-        except (asyncio.TimeoutError, aiohttp.ClientError):
-            _LOGGER.error("Error fetching %s", image_url)
-            return None
-
-        image = yield from response.read()
-
-        return image
+        return self._surveillance.get_camera_image(self._camera_id)
 
     @asyncio.coroutine
     def handle_async_mjpeg_stream(self, request):
         """Return a MJPEG stream image response directly from the camera."""
-        streaming_url = SYNO_API_URL.format(
-            self._synology_url, WEBAPI_PATH, self._streaming_path)
-
-        streaming_payload = {
-            'api': STREAMING_API,
-            'method': 'Stream',
-            'version': '1',
-            'cameraId': self._camera_id,
-            'format': 'mjpeg'
-        }
-        stream_coro = self._websession.get(
-            streaming_url, params=streaming_payload)
+        streaming_url = self._camera.video_stream_url
+        stream_coro = self._websession.get(streaming_url)
 
         yield from async_aiohttp_proxy_web(self.hass, request, stream_coro)
 
     @property
     def name(self):
         """Return the name of this device."""
+        return self._camera.name
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return self._camera.is_recording
+
+    def should_poll(self):
+        """Update the recording state periodically."""
+        return True
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the status of the camera."""
+        self._surveillance.update()
+        self._camera = self._surveillance.get_camera(self._camera.camera_id)
+        self._motion_setting = self._surveillance.get_motion_setting(
+            self._camera.camera_id)
+        self.is_streaming = self._camera.is_enabled
+
+    @property
+    def motion_detection_enabled(self):
+        """Return the camera motion detection status."""
+        return self._motion_setting.is_enabled
+
+    def enable_motion_detection(self):
+        """Enable motion detection in the camera."""
+        self._surveillance.enable_motion_detection(self._camera_id)
+
+    def disable_motion_detection(self):
+        """Disable motion detection in camera."""
+        self._surveillance.disable_motion_detection(self._camera_id)
+
+
+BASE_API_INFO = {
+    'auth': {
+        'name': 'SYNO.API.Auth',
+        'version': 2
+    },
+    'camera': {
+        'name': 'SYNO.SurveillanceStation.Camera',
+        'version': 1
+    },
+    'camera_event': {
+        'name': 'SYNO.SurveillanceStation.Camera.Event',
+        'version': 1
+    },
+    'video_stream': {
+        'name': 'SYNO.SurveillanceStation.VideoStream',
+        'version': 1
+    },
+}
+
+API_NAMES = map(lambda api: api['name'], BASE_API_INFO.values())
```

Comment:
Use a list comprehension instead.

Reply:
Sorry I'm new to python world so hopefully the change I made is what you meant: 
https://github.com/snjoetw/py-synology/blob/master/synology/api.py#L25